server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1627,39 +1665,6 @@ private SearchSourceBuilder parseXContent(
         }
 
         knnSearch = knnBuilders.stream().map(knnBuilder -> knnBuilder.build(size())).collect(Collectors.toList());
-
-        if (retrieverBuilder != null) {
```

Comment:
Now that we have moved this validation step outside of the main parsing into a separate `validate` method, is there a scenario where we might end up not running these checks at all, or should we be covered in all cases by the `SearchRequest` modifications?

Reply:
That's a good question. We have the same problem with the other validations that I moved here. These validations are missing from action request that uses a search source builder internally. I don't think it's an issue though since these actions use the search action internally so the validation is just delayed to the actual execution of the search.